### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by Co0sh**
This adds `http-errors` package for easier error handling and modifies resource hooks to display error messages in the `notice` filed of the response.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,11 +75,13 @@
     "@babel/preset-typescript": "^7.8.3",
     "@babel/register": "^7.8.3",
     "@carbon/icons-react": "^10.8.1",
+    "@types/http-errors": "^1.6.3",
```
Comment (wojtek-krysiak) [inline package.json]: I think that both @types/http-errors and @types/react (my bad) should go devdeps. What do you think?

File: src/frontend/hooks/use-resource-edit.ts [hunk 2]
```diff
@@ -69,11 +69,11 @@ const useResourceEdit = (
           setLoading(false)
         }
       })
-      .catch(() => {
+      .catch((error) => {
         setLoading(false)
         onNotice({
-          message:
-            'There was an error updating record, Check out console to see more information.',
+          message: error.response?.data.notice
```
Comment (wojtek-krysiak) [inline src/frontend/hooks/use-resource-edit.ts]: in the latest release I added i18n, so all errors goes throught tranlsateMessage function().

**Comment by piotrwilczynski88** [comment]
# [Codecov](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=h1) Report
> Merging [#309](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=desc) into [v2.0](https://codecov.io/gh/SoftwareBrothers/admin-bro/commit/d69dc07662b8bd3e707ae0caac6deb6a82b20e55&el=desc) will **increase** coverage by `7.85%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##             v2.0     #309      +/-   ##
==========================================
+ Coverage   51.18%   59.04%   +7.85%     
==========================================
  Files         186      235      +49     
  Lines        2454     3777    +1323     
  Branches      408      565     +157     
==========================================
+ Hits         1256     2230     +974     
- Misses       1099     1410     +311     
- Partials       99      137      +38     
```


| [Impacted Files](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/backend/actions/list-action.ts](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2JhY2tlbmQvYWN0aW9ucy9saXN0LWFjdGlvbi50cw==) | `41.66% <0.00%> (-58.34%)` | :arrow_down: |
| [...ponents/design-system/atoms/illustrations/moon.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL21vb24udHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...nents/design-system/atoms/illustrations/launch.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2xhdW5jaC50c3g=) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...ents/design-system/atoms/illustrations/folders.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2ZvbGRlcnMudHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [.../design-system/atoms/illustrations/file-search.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2ZpbGUtc2VhcmNoLnRzeA==) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [.../design-system/atoms/illustrations/github-logo.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2dpdGh1Yi1sb2dvLnRzeA==) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...esign-system/atoms/illustrations/adminbro-logo.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2FkbWluYnJvLWxvZ28udHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...sign-system/atoms/illustrations/document-check.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2RvY3VtZW50LWNoZWNrLnRzeA==) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...tem/atoms/illustrations/software-brothers-logo.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL3NvZnR3YXJlLWJyb3RoZXJzLWxvZ28udHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...rontend/components/property-type/datetime/show.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvcHJvcGVydHktdHlwZS9kYXRldGltZS9zaG93LnRzeA==) | `50.00% <0.00%> (-50.00%)` | :arrow_down: |
| ... and [140 more](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=footer). Last update [d69dc07...9ea6d74](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by piotrwilczynski88** [comment]
# [Codecov](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=h1) Report
> Merging [#309](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=desc) into [v2.0](https://codecov.io/gh/SoftwareBrothers/admin-bro/commit/fcaff4bb6ca6eb019d6ce487b2990608a3434663&el=desc) will **increase** coverage by `7.84%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##             v2.0     #309      +/-   ##
==========================================
+ Coverage   51.18%   59.02%   +7.84%     
==========================================
  Files         186      235      +49     
  Lines        2454     3795    +1341     
  Branches      408      569     +161     
==========================================
+ Hits         1256     2240     +984     
- Misses       1099     1417     +318     
- Partials       99      138      +39     
```


| [Impacted Files](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/backend/actions/list-action.ts](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2JhY2tlbmQvYWN0aW9ucy9saXN0LWFjdGlvbi50cw==) | `41.66% <0.00%> (-58.34%)` | :arrow_down: |
| [...ponents/design-system/atoms/illustrations/moon.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL21vb24udHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...nents/design-system/atoms/illustrations/launch.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2xhdW5jaC50c3g=) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...ents/design-system/atoms/illustrations/folders.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2ZvbGRlcnMudHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [.../design-system/atoms/illustrations/file-search.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2ZpbGUtc2VhcmNoLnRzeA==) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [.../design-system/atoms/illustrations/github-logo.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2dpdGh1Yi1sb2dvLnRzeA==) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...esign-system/atoms/illustrations/adminbro-logo.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2FkbWluYnJvLWxvZ28udHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...sign-system/atoms/illustrations/document-check.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL2RvY3VtZW50LWNoZWNrLnRzeA==) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...tem/atoms/illustrations/software-brothers-logo.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvZGVzaWduLXN5c3RlbS9hdG9tcy9pbGx1c3RyYXRpb25zL3NvZnR3YXJlLWJyb3RoZXJzLWxvZ28udHN4) | `42.85% <0.00%> (-57.15%)` | :arrow_down: |
| [...rontend/components/property-type/datetime/show.tsx](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree#diff-c3JjL2Zyb250ZW5kL2NvbXBvbmVudHMvcHJvcGVydHktdHlwZS9kYXRldGltZS9zaG93LnRzeA==) | `50.00% <0.00%> (-50.00%)` | :arrow_down: |
| ... and [142 more](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=footer). Last update [fcaff4b...3be4a1e](https://codecov.io/gh/SoftwareBrothers/admin-bro/pull/309?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by maurocen** [comment]
@wojtek-krysiak I think the changes requested have been made.

**Comment by Co0sh** [comment]
Closing since it's no longer relevant.

**Pull request closed without merging**